src/dvclive/live.py

```diff
@@ -457,6 +465,7 @@ def log_image(
         self,
         name: str,
         val: Union[np.ndarray, matplotlib.figure.Figure, PIL.Image, StrPath],
+        annotations: Optional[Dict] = None,
```

Comment:
Should we include a more specific type here to generate better hints as we discussed @AlexandreKempf?

We also need to update the docstring to include `annotations`. Is it too much to fit the structure of the annotations into the docstring?

Reply:
:+1:  I added the docstring, you were right, it was missing. Tell me if this is too much. I feel like the length of the docstring is fine. 

For better typing, I looked at several options and I picked the one that I preferred:
- Pydantic, data class, attrs ... Not a huge fan since it introduces a new object the user needs to import and use. ex: `log_image(path, image, annotations=Annotations(boxes= ..., labels=..., scores=..., format=....))`. If we are OK with that, it is probably the most elegant solution otherwise 
- TypedDict all the way. I tried to replace all the annotations dictionaries in `plots/annotations.py`. If we do that, mypy refuses to use constant names (`BOXES_NAMES,` `LABELS_NAME,`...) to access the typedict keys. It is not a problem, we could use "boxes" instead of a constant, and mypy would warn use if we try to access a field that is not valid. The problem is that if we change the name of a field, it is slightly annoying to replace the field everywhere (main code + log warning + tests). 
- **TypedDict only for the user (in `live.py`) -> the option I picked. A duplicated schema exists between what is written in `live.py`  and `annotations.py`. That is one downside, but I think it is the least annoying downside of all.**
- Dict as it was during your comment. The problem as you mention is that it is a bit vague for the user that uses it for the first time. 

What do you think? Do you see any other option?

Reply:
I'll be using pydantic